packages/@sanity/cli/src/actions/init-project/initProject.ts

```diff
@@ -202,9 +238,163 @@ export default async function initSanity(
+          ':configPath:',
+          new Array(embeddedStudioPath.slice(workDir.length).split('/').length - 1)
+            .join('../')
+            .concat('sanity.config')
+        )
+      )
+
+      const sanityConfigPath = path.join(workDir, 'sanity.config.'.concat(fileEnding))
+      await writeOrOverwrite(
+        sanityConfigPath,
+        sanityConfigTemplate
+          .replace(':route:', embeddedStudioPath.slice(workDir.length).replace('src/', ''))
+          .replace(':basePath:', studioPath)
+      )
+    }
+
+    const sanityCliPath = path.join(workDir, 'sanity.cli.'.concat(fileEnding))
+    await writeOrOverwrite(sanityCliPath, sanityCliTemplate)
+
+    // write sanity folder files
+    const recursivelyWriteFiles = async (
+      files: Record<string, string | Record<string, string>>,
+      folderPath?: string
+    ) => {
+      for (const [filePath, content] of Object.entries(files)) {
+        if (filePath.includes('.') && typeof content === 'string') {
+          await writeOrOverwrite(
+            path.join(workDir, 'sanity', folderPath || '', filePath.concat(fileEnding)),
+            content
+          )
+        } else {
+          await mkdirSync(path.join(workDir, 'sanity', filePath), {recursive: true})
+          if (typeof content === 'object') {
+            await recursivelyWriteFiles(content, filePath)
+          }
+        }
+      }
+    }
+
+    // ask what kind of schema setup the user wants
+    const templateToUse = await promptForNextTemplate(prompt)
+
+    const spinner = await recursivelyWriteFiles(sanityFolder(useTypeScript, templateToUse)).then(
+      () => {
+        return context.output.spinner('Installing Sanity dependencies').start()
+      }
+    )
+
+    // set tsconfig.json target to ES2017
+    const tsConfigPath = path.join(workDir, 'tsconfig.json')
+    if (useTypeScript && existsSync(tsConfigPath)) {
+      const tsConfig = JSON.parse(readFileSync(tsConfigPath, 'utf8'))
+      tsConfig.compilerOptions.target = 'ES2017'
+      await writeFileSync(tsConfigPath, JSON.stringify(tsConfig, null, 2))
+    }
+
+    // install sanity deps
+    await exec('npm install next-sanity@4 @sanity/vision@3 sanity@3 @sanity/image-url@1', {
+      cwd: workDir,
+    })
+      .on('exit', async () => {
+        spinner.succeed()
+
+        const appendEnv = await promptForAppendEnv(prompt, envFilename)
+
+        if (appendEnv) {
+          await createOrAppendEnvVars(envFilename, detectedFramework, {
+            noLogs: true,
+          })
+        }
+
+        print(
+          `\n${chalk.green(
+            'Success!'
+          )} Your Sanity configuration files has been added to this project`
+        )
+
+        // eslint-disable-next-line no-process-exit
+        process.exit(0)
+      })
+      .on('stderr', (err) => Promise.reject(err))
+
+    return
+  }
+
+  async function writeOrOverwrite(
+    // eslint-disable-next-line @typescript-eslint/no-shadow
+    path: string,
```

Comment:
How about just naming it `filePath` or similar instead? Shadowing can be a little confusing